### PR TITLE
Added setup fixture to test "get_cli_cmd"

### PR DESCRIPTION
**Pull request by AviadP**
- [x]  Fix initialization error in test
- [x] Fix parsing errors

### CODE REVIEW

File: ocs_ci/helpers/odf_cli.py [hunk 1]
```diff
@@ -18,23 +17,40 @@
...
+
+        if not self.check_odf_cli_binary():
+            raise RuntimeError(
+                f"Failed to retrieve and set up ODF CLI binary at {self.local_cli_path}"
+            )
+
+        log.info(f"ODF CLI binary is ready at {self.local_cli_path}")
 
     def _validate_odf_cli_support(self):
         if self.semantic_version < VERSION_4_15:
```
Comment (shyRozen) [inline ocs_ci/helpers/odf_cli.py:56]: Maybe we can do it with a marker so it will be skipped and not failed?

Comment (AviadP) [reply ocs_ci/helpers/odf_cli.py:56]: sure

**Review by OdedViner — DISMISSED** [review]

**Review by OdedViner — APPROVED** [review]

File: tests/functional/odf-cli/test_get_commands.py [hunk 2]
```diff
@@ -1,24 +1,45 @@
...
 
 @tier1
 @brown_squad
+@skipif_ocs_version("<4.15")
 class TestGetCommands:
-    def test_download_binary(self):
-        self.odf_cli_retriever = ODFCLIRetriever()
-        self.odf_cli_retriever.retrieve_odf_cli_binary()
-        self.odf_cli_runner = ODFCliRunner()
+    @pytest.fixture(scope="function", autouse=True)
```
Comment (ebenahar) [inline tests/functional/odf-cli/test_get_commands.py:16]: From what I know, `autouse=True` means this fixture will run for every function scoped test iteration. Do we really need this fixture to run before all of ocs-ci test cases?

**Review by ebenahar — CHANGES_REQUESTED** [review]

File: tests/functional/odf-cli/test_get_commands.py (hunk 2 above)
Comment (ebenahar) [reply tests/functional/odf-cli/test_get_commands.py:16]: please ignore. For some reason I thought this is a fixture in conftest.py

File: tests/functional/odf-cli/test_get_commands.py [hunk 3]
```diff
@@ -1,24 +1,45 @@
...
+    def odf_cli_setup(self):
```
Comment (ebenahar) [inline tests/functional/odf-cli/test_get_commands.py:17]: shouldn't ODF CLI binary be validated as part of post deployment or maybe as a session scope fixture?

Comment (AviadP) [reply tests/functional/odf-cli/test_get_commands.py:17]: im not sure about post deployment, but it can be session scope. i will test it as session

**Review by OdedViner — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**